### PR TITLE
Session Config

### PR DESCRIPTION
**Pull request by mostafakhudair**
Split session configuration from `Config::App` to a new config class `Config::Session`

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPdocs
- [x] Unit testing, with >80% coverage
- [x] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Comment by mostafakhudair** [comment]
unfamiliar fail

File: system/Session/Handlers/FileHandler.php [hunk 1]
```diff
@@ -48,13 +48,6 @@ class FileHandler extends BaseHandler
      */
     protected $fileNew;
 
-    /**
-     * Whether IP addresses should be matched.
-     *
-     * @var bool
-     */
-    protected $matchIP = false;
```
Comment (paulbalandan) [inline system/Session/Handlers/FileHandler.php:56]: Why is this removed? Is this duplicated by BaseHandler?

File: user_guide_src/source/libraries/sessions.rst [hunk 2]
```diff
@@ -428,26 +428,26 @@ careful configuration must be done. Please take your time to consider
...
 
 You'll find the following Session related preferences in your
-**app/Config/App.php** file:
+**app/Config/Session.php** file:
 
 ============================== ============================================ ================================================= ============================================================================================
 Preference                     Default                                      Options                                           Description
 ============================== ============================================ ================================================= ============================================================================================
-**sessionDriver**              CodeIgniter\\Session\\Handlers\\FileHandler  CodeIgniter\\Session\\Handlers\\FileHandler       The session storage driver to use.
+**handler**              CodeIgniter\\Session\\Handlers\\FileHandler  CodeIgniter\\Session\\Handlers\\FileHandler       The session storage driver to use.
```
Comment (paulbalandan) [inline user_guide_src/source/libraries/sessions.rst:436]: Table is malformed maybe due to spacing.

File: system/Session/Handlers/MemcachedHandler.php [hunk 3]
```diff
@@ -45,9 +46,18 @@ class MemcachedHandler extends BaseHandler
...
+     * @deprecated
      */
     protected $sessionExpiration = 7200;
 
+    /**
+     * Number of seconds until the session ends.
+     *
+     * @var int
+     */
+    protected $lifetime = 7200;
```
Comment (paulbalandan) [inline system/Session/Handlers/MemcachedHandler.php:59]: I think instead of replicating across all other handlers you can define this instead in BaseHandler. Even if there are handlers not using this, at least it is there available.

File: system/Session/Session.php [hunk 4]
```diff
@@ -35,17 +36,88 @@ class Session implements SessionInterface
      */
     protected $driver;
 
+    /**
+     * The session storage driver to use.
+     *
+     * @var string
+     */
+    protected $handler = FileHandler::class;
```
Comment (paulbalandan) [inline system/Session/Session.php:44]: I would be in favor of marking these new props as private to reduce BC concerns in the future. This also promotes encapsulation. If a user will find this implementation as lacking, they can abstract one using the `SessionInterface` instead of extending this and complaining this and that. But this is just my preference.

**Comment by MGatner** [comment]
@mostafakhudair We would like to do another release soon, and I would love to see this as part of it. Could you address Paul's requests? I will make it a point to do my full review soon.

**Comment by kenjis** [comment]
It seems the base branch of this PR must be changed to `develop`.

**Comment by kenjis** [comment]
It seems the base branch of this PR must be changed to `develop`.

**Comment by kenjis** [comment]
@MGatner What is the version number of the next release? 4.1.5? 4.2.0?

**Comment by mostafakhudair** [comment]
@MGatner It will be ready tomorrow

File: system/Session/Handlers/FileHandler.php (hunk 1 above)
Comment (mostafakhudair) [reply system/Session/Handlers/FileHandler.php:56]: Yes it is

**Comment by mostafakhudair** [comment]
> @MGatner What is the version number of the next release? 4.1.5? 4.2.0?

@MGatner  should this go with develop or stick with 4.2 branch

File: system/Session/Session.php (hunk 4 above)
Comment (kenjis) [reply system/Session/Session.php:44]: I understand your preference.

But the way to extend the core classes is described in the user guide.
https://codeigniter4.github.io/CodeIgniter4/extending/core_classes.html#extending-core-classes
And CI users have been using it a lot for a long time.

Making properties as private is big change. I don't think it's easy to decide.

File: app/Config/Session.php [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+     * --------------------------------------------------------------------------
+     * Session Life Time
+     * --------------------------------------------------------------------------
+     *
+     * The number of SECONDS you want the session to last.
+<<<<<<< HEAD
+     * Set to `0` means expire when the browser is closed.
+=======
+     * Setting to 0 (zero) means expire when the browser is closed.
+>>>>>>> 8d8c82bba537670571013bfdbd155c7a4e91b2aa
```
Comment (kenjis) [inline app/Config/Session.php:47]: There is a conflict.

**Comment by MGatner** [comment]
@mostafakhudair Please redirect to `develop` and rebase. Sorry for the bouncing back and forth - this should never happen again, was an odd artifact of restyling the codebase. 

@kenjis Next version will be 4.1.5, including everything that was on the 4.2 branch (minus `Publisher`), and releasing as soon as this PR is ready 😉

**Comment by MGatner** [comment]
@mostafakhudair Are we getting this back? Maybe in a fresh PR? Keen on this.

**Comment by mostafakhudair** [comment]
@MGatner  new clean PR created #5088
